runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -76,23 +77,21 @@ public int size() {
      * This class encapsulates these two longs into one Id and add rules on comparability.
      */
     public static class CorfuRecordId implements Comparable<CorfuRecordId> {
-        private final UUID id;
-        public CorfuRecordId(long ordering, long uniqueId) {
-            this.id = new UUID(ordering, uniqueId);
-        }
+        @Setter
```

Comment:
is it possible to get rid of setters?

Reply:
no, it breaks SMR if we create a new object - we *must* modify the exact value that was added to the table's hash map.

Reply:
not sure about SMR, but if you are going to change these fields and use them to calculate hashCode then it is very dangerous operation you can break HashMap by that.
I mean if you change those fields after you add it to any java data structure like HashMap, Queue and so on, you will not be able to find that entry in the data structure